lib/spack/docs/workflows.rst

```diff
@@ -1090,6 +1090,211 @@ or filesystem views.  However, it has some drawbacks:
+ever want to add more packages, you can either use ``spack add`` or manually
+edit this file.
+
+We also need to change the ``concretization:`` option. By default, Spack
+concretizes each spec *separately*, allowing multiple versions of the same
+package to coexist. Since we want a single consistent environment, we want to
+concretize all of the specs *together*.
+
+Here is what your ``spack.yaml`` looks like with these new settings, and with
+some of the sections we don't plan on using removed:
+
+.. code-block:: yaml
+
+   spack:
+     specs:
+     - bash
+     - 'python@3:'
+     - py-numpy
+     - py-scipy
+     - py-matplotlib
+     view: /Users/me/spack/var/spack/environments/myenv/.spack-env/view
+     concretization: together
+
+""""""""""""""""
+Symlink location
+""""""""""""""""
+
+In the ``spack.yaml`` file above, you'll notice that by default, Spack symlinks
+all installations to ``/Users/me/spack/var/spack/environments/myenv/.spack-env/view``.
+You can actually change this to any directory you want. For example, Homebrew
+uses ``/usr/local``, while Conda uses ``/Users/me/anaconda``. In order to access
+files in these locations, you need to update ``PATH`` and other environment variables
+to point to them. Activating the Spack environment does this automatically, but
+you can also manually set them in your ``.bashrc``.
+
+.. warning::
+
+   There are several reasons why you shouldn't use ``/usr/local``:
+
+   1. If you are on macOS 10.11+ (El Capitan and newer), Apple makes it hard
+      for you. You may notice permissions issues on ``/usr/local`` due to their
+      `System Integrity Protection <https://support.apple.com/en-us/HT204899>`_.
+      By default, users don't have permissions to install anything in ``/usr/local``,
+      and you can't even change this using ``sudo chown`` or ``sudo chmod``.
+   2. Other package managers like Homebrew will try to install things to the
+      same directory. If you plan on using Homebrew in conjunction with Spack,
+      don't symlink things to ``/usr/local``.
+   3. If you are on a shared workstation, or don't have sudo priveleges, you
+      can't do this.
+
+   If you still want to do this anyway, there are several ways around SIP.
+   You could disable SIP by booting into recovery mode and running
+   ``csrutil disable``, but this is not recommended, as it can open up your OS
+   to security vulnerabilities. Another technique is to run ``spack concretize``
+   and ``spack install`` using ``sudo``. This is also not recommended.
+
+   The safest way I've found is to create your installation directories using
+   sudo, then change ownership back to the user like so:
+
+   .. code-block:: bash
+
+      for directory in .spack bin include lib share
+      do
+          sudo mkdir /usr/local/$directory
+          sudo chown $USER:staff /usr/local/$directory
+      done
+
+   But again, it's safer just to use the default symlink location.
```

Comment:
I'm obviously ignoring this, so I'll let you know if it causes any trouble.

Reply:
I'm still thinking about this. The intent of this workflow is specifically how to replace Homebrew/Conda with Spack. There are other sections of the docs already devoted to more generic usage of Spack Environments and flat installation hierarchies:

- [Spack Environments](https://spack.readthedocs.io/en/latest/environments.html)
- [Environments Tutorial](https://spack.readthedocs.io/en/latest/tutorial_environments.html)
- [Spack View Workflow](https://spack.readthedocs.io/en/latest/workflows.html#cmd-spack-view)

I do like your suggestions for enumerating through the list of env vars you need to set to contrast why you might want to use `/usr/local`, but you can also just `spack env activate myenv` and it will set all of these automatically for you, much like `module load` would.